doc/architecture.md

```diff
@@ -0,0 +1,136 @@
+Architecture
+============
+
+This section describes how a Management Pack works, and how it interacts with the VMware 
+Aria Operations Collector.
+
+## Management Pack
+A management pack consists of content, metadata, and an adapter.
+
+* Content includes Dashboards, Reports, Icons, Symptoms, Alerts, Traversals, others.
+* Metadata includes the describe.xml file that describes an adapter, and the 
+  `manifest.txt` file that describes the management pack as a whole.
+* The Adapter is the code that performs the collection and sends the data to VMware Aria 
```

Comment:
Might be worth pointing out that the Adapter code is the main part that a developer needs to write.